pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/reporter/TopBundleLoadDataReporter.java

```diff
@@ -0,0 +1,88 @@
+ * under the License.
+ */
+package org.apache.pulsar.broker.loadbalance.extensions.reporter;
+
+import java.util.concurrent.CompletableFuture;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.broker.PulsarService;
+import org.apache.pulsar.broker.loadbalance.extensions.data.TopBundlesLoadData;
+import org.apache.pulsar.broker.loadbalance.extensions.models.TopKBundles;
+import org.apache.pulsar.broker.loadbalance.extensions.store.LoadDataStore;
+
+@Slf4j
+public class TopBundleLoadDataReporter implements LoadDataReporter<TopBundlesLoadData> {
+
+    private final PulsarService pulsar;
+
+    private final String lookupServiceAddress;
+
+    private final LoadDataStore<TopBundlesLoadData> bundleLoadDataStore;
+
+    private final TopKBundles topKBundles;
+
+    private long lastBundleStatsUpdatedAt;
+
+    public TopBundleLoadDataReporter(PulsarService pulsar,
+                                     String lookupServiceAddress,
+                                     LoadDataStore<TopBundlesLoadData> bundleLoadDataStore) {
+        this.pulsar = pulsar;
+        this.lookupServiceAddress = lookupServiceAddress;
+        this.bundleLoadDataStore = bundleLoadDataStore;
+        this.lastBundleStatsUpdatedAt = 0;
+        this.topKBundles = new TopKBundles();
+    }
+
+    @Override
+    public TopBundlesLoadData generateLoadData() {
+
+        var pulsarStats = pulsar.getBrokerService().getPulsarStats();
+        TopBundlesLoadData result = null;
+        synchronized (pulsarStats) {
+            var pulsarStatsUpdatedAt = pulsarStats.getUpdatedAt();
+            if (pulsarStatsUpdatedAt > lastBundleStatsUpdatedAt) {
+                var bundleStats = pulsar.getBrokerService().getBundleStats();
+                double percentage = pulsar.getConfiguration().getLoadBalancerBundleLoadReportPercentage();
+                int topk = Math.max(1, (int) (bundleStats.size() * percentage / 100.0));
+                topKBundles.update(bundleStats, topk);
+                lastBundleStatsUpdatedAt = pulsarStatsUpdatedAt;
+                result = topKBundles.getLoadData();
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public CompletableFuture<Void> reportAsync(boolean force) {
+        try {
+            var topBundlesLoadData = generateLoadData();
+            if (topBundlesLoadData != null || force) {
+                return this.bundleLoadDataStore.pushAsync(lookupServiceAddress, topKBundles.getLoadData())
+                        .exceptionally(e -> {
+                            log.error("Failed to report top-bundles load data.", e);
+                            return null;
+                        });
+            } else {
+                return CompletableFuture.completedFuture(null);
+            }
+        } catch (Throwable e) {
+            log.error("Failed to report top-bundles load data.", e);
+            return CompletableFuture.failedFuture(e);
```

Comment:
Why do we need to catch exception ?

Reply:
This is to wrap any unexpected exception and continue the next run on the scheduler thread.

Reply:
yes, but seems no error here.

Reply:
Removed this try-catch, and added try-catch to the scheduler.